### ISSUE TITLE
Use Vetur's html parser instead of vue-eslint-parser for template diagnostic

### ISSUE THREAD

**Issue by ktsn** [vuejs/vetur#1205]

**Comment by octref** [comment]
I'm not sure if we still should do this. For parsing the JS expressions in interpolations you still need eslint-plugin-vue, correct?

Also maybe we should try to generate sourcemap from our template -> JS transformation.

**Comment by ktsn** [comment]
Not really. We rely on TypeScript parser for expressions in template. That's why we need to pass template code to transform functions.

Actually I don't have strong opinion on this as I just want to avoid to use different approaches for one purpose (template parsing). 

And I just saw [this comment](https://github.com/vuejs/vetur/blob/master/server/src/modes/template/services/vueInterpolationCompletion.ts#L6). If that means you want to switch template parser to vue-eslint-parser entirely in the future, I think we should keep using it.

**Comment by octref** [comment]
@ktsn I meant for things like `VFilterSequenceExpression` etc, I don't think you can do those with TS and our HTML parser alone, correct?

Of course, for general expressions TS is enough.

**Comment by ktsn** [comment]
Ah, I see. Yeah we need to extend our html parser to deal with it...

**Comment by ktsn** [comment]
Let's keep going with vue-eslint-parser.
